File: packages/notesprout_core/lib/models/base_object.dart
abstract class BaseObject {
  final String id;
  final String? parentId;
  final String type;
  final String? subtype;
  final DateTime createdAt;
  final DateTime updatedAt;
  final DateTime? deletedAt;

  const BaseObject({
    required this.id,
    this.parentId,
    required this.type,
    this.subtype,
    required this.createdAt,
    required this.updatedAt,
    this.deletedAt,
  });
}

File: apps/notesprout_android/app/src/main/kotlin/com/notesprout/android/data/BaseObject.kt
package com.notesprout.android.data

data class BaseObject(
    val id: String,
    val parentId: String,
    val pluginId: String,
    val boundingBox: BoundingBox,
    val order: Int,
    val createdAt: Long,
    val updatedAt: Long,
    val deletedAt: Long?,
    val syncVersion: Long,
    val data: String
)
